packages/control-group/src/index.tsx

```diff
@@ -0,0 +1,241 @@
+        >
+          {item.label}
+        </Radio>
+      )
+    case 'checkbox-horizontal':
+    case 'checkbox-vertical':
+      return (
+        <Checkbox
+          {...renderProps}
+          appearance={controlAppearance as CheckboxAppearanceType}
+          onChange={() => safeInvoke(handleOnChange, item.value)}
+        >
+          {item.label}
+        </Checkbox>
+      )
+    case 'button':
+    default:
+      return (
+        <ControlButton
+          {...renderProps}
+          appearance={controlAppearance as ButtonAppearanceType}
+          onClick={() => safeInvoke(handleOnChange, item.value)}
+          activeGroup={Boolean(
+            Array.isArray(groupValue) ? groupValue.length : groupValue
+          )}
+          isOpen={isOpen}
+        >
+          {isOutline ? <Strong>{item.label}</Strong> : item.label}
+        </ControlButton>
+      )
+  }
+}
+
+const ControlGroupFC: React.FC<ControlGroupProps> = props => {
+  const {height, controlAppearance, groupAppearance} = useDefaults(
+    'controlGroup',
+    props,
+    {
+      height: 40,
+      controlAppearance: 'default' as ControlAppearanceType,
+      groupAppearance: 'button' as ControlGroupAppearanceType
+    }
+  )
+  const {
+    items,
+    onChange,
+    value,
+    layout,
+    textAlign,
+    disabled,
+    invalid,
+    visibleCount
+  } = props
+
+  const [controlsNumber, setControlsNumber] = React.useState(visibleCount || 0)
+  const [isOpen, setIsOpen] = React.useState(false)
+  const hasMoreButton =
+    controlAppearance === 'outline' && groupAppearance === 'button'
+  const wrapperNode = React.useRef<HTMLDivElement>(null)
+  const moreButtonNode = React.useRef<HTMLButtonElement>(null)
+
+  React.useEffect(() => {
+    if (
+      wrapperNode &&
+      wrapperNode.current &&
+      !controlsNumber &&
+      !visibleCount
+    ) {
+      const {clientWidth, children: listChildren} = wrapperNode.current
+      let widthSum = moreButtonNode.current
+        ? moreButtonNode.current.clientWidth
+        : 0
+      let childCount = 0
+
+      Array.from(listChildren).forEach(child => {
+        if (widthSum + child.scrollWidth <= clientWidth) {
+          widthSum += child.scrollWidth
+          childCount++
+        }
+      })
+      setControlsNumber(childCount)
+    }
+  }, [wrapperNode])
+
+  const handleMoreClick = (e: React.MouseEvent) => {
+    const moreButton = moreButtonNode.current
+    if (!moreButton || !moreButton.contains(e.target as Node)) {
+      setIsOpen(false)
+    }
+  }
+
+  return (
+    <ControlGroupWrapper
+      ref={wrapperNode}
+      groupAppearance={groupAppearance}
+      controlAppearance={controlAppearance}
+      childrenAmount={items.length}
+      layout={layout}
+      isOpen={isOpen}
+      visibleItemsCount={controlsNumber}
+      onClick={handleMoreClick}
+    >
+      {items.map(item => (
+        <Control
+          item={item}
+          onChange={onChange}
```

Comment:
there are two weird behaviors related to more/clicking an option:
1) When only three options are set (by visibleCount), it's last one displayed instead of third
2) "More" button isn't visible until clicking an option

![ezgif-2-68b31226c82a](https://user-images.githubusercontent.com/8265332/70214607-08b86400-173c-11ea-9422-fd388c5c2f7d.gif)

Reply:
fixed